File: Medium/ Code  Testcase  1877. Minimize Maximum Pair Sum in Array/1877. Minimize Maximum Pair Sum in Array.py
from typing import List

class Solution:
    def minPairSum(self, nums: List[int]) -> int:
        nums.sort()

        max_pair_sum = 0
        n = len(nums)

        for i in range(n // 2):
            max_pair_sum = max(max_pair_sum, nums[i] + nums[n - i - 1])

        return max_pair_sum

File: Medium/1877. Minimize Maximum Pair Sum in Array/1877. Minimize Maximum Pair Sum in Array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minPairSum = function (nums) {
    nums.sort((a, b) => a - b);

    let maxPairSum = 0;
    const n = nums.length;

    for (let i = 0; i < n / 2; i++) {
        maxPairSum = Math.max(maxPairSum, nums[i] + nums[n - i - 1]);
    }

    return maxPairSum;
};
